dot/rpc/modules/state.go

```diff
@@ -233,10 +237,24 @@ func (sm *StateModule) GetChildStorageSize(r *http.Request, req *StateChildStora
 	return nil
 }
 
-// GetKeys isn't implemented properly yet.
-func (sm *StateModule) GetKeys(r *http.Request, req *StateStorageKeyRequest, res *StateStorageKeysResponse) error {
-	// TODO implement change storage trie so that block hash parameter works (See issue #834)
-	return nil
+// GetKeysPaged Returns the keys with prefix with pagination support.
+func (sm *StateModule) GetKeysPaged(r *http.Request, req *StateStorageKeyRequest, res *StateStorageKeysResponse) error {
+	keys, err := sm.storageAPI.Keys(req.Block)
+	resCount := uint32(0)
+	for _, k := range keys {
+		fKey := fmt.Sprintf("0x%x", k)
+		if strings.HasPrefix(fKey, req.Prefix) &&
```

Comment:
the trie has a method `GetKeysWithPrefix`, you could add that to the StorageState and use that